src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfaceSecurity.cs

```diff
@@ -190,6 +143,59 @@ class ConsoleHostUserInterface : System.Management.Automation.Host.PSHostUserInt
 
             return cred;
         }
+
+        private bool ComparePasswords(SecureString password, SecureString retypedPassword)
+        {
+            IntPtr passwordBstr = IntPtr.Zero;
+            IntPtr retypedPasswordBstr = IntPtr.Zero;
+            int match = 0;
+            try
+            {
+                passwordBstr = Marshal.SecureStringToBSTR(password);
+                retypedPasswordBstr = Marshal.SecureStringToBSTR(retypedPassword);
+                int passwordLength = Marshal.ReadInt32(passwordBstr, -4);
+                int confirmPasswordLength = Marshal.ReadInt32(retypedPasswordBstr, -4);
```

Comment:
See answer in https://stackoverflow.com/questions/4502676/c-sharp-compare-two-securestrings-for-equality
It looks more reliable: 
```c#
        ss_bstr1_ptr = Marshal.SecureStringToBSTR(secureString1);
        ss_bstr2_ptr = Marshal.SecureStringToBSTR(secureString2);

        String str1 = Marshal.PtrToStringBSTR(ss_bstr1_ptr);
        String str2 = Marshal.PtrToStringBSTR(ss_bstr2_ptr);

        return str1.Equals(str2);
``